src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ValueTask.cs

```diff
@@ -56,8 +56,8 @@ namespace System.Threading.Tasks
     [StructLayout(LayoutKind.Auto)]
     public readonly struct ValueTask : IEquatable<ValueTask>
     {
-        /// <summary>A task canceled using `new CancellationToken(true)`.</summary>
-        private static readonly Task s_canceledTask = Task.FromCanceled(new CancellationToken(canceled: true));
+        /// <summary>A task canceled using `new CancellationToken(true)`. Lazily created only when first needed.</summary>
+        private static Task? s_canceledTask;
```

Comment:
Nit: Consider annotating the write to this field as _volatile_. I know it's an implementation detail of the runtime right now that all all ref-type writes to the heap have implicit release semantics. But IMO it's best practice to make this explicit, and it's a bit more future-proof in case we ever get a runtime that doesn't have those semantics.

Reply:
Yup.  Both read and write technically need to be from an ECMA perspective.  I'll just make the fields volatile.